core/src/main/scala/com/rallyhealth/vapors/core/algebra/HExprList.scala

```diff
@@ -0,0 +1,29 @@
+package com.rallyhealth.vapors.core.algebra
+
+import cats.syntax.all._
+import cats.{Functor, Semigroupal}
+import shapeless.{::, HList, HNil}
+
+// TODO: How to convert this back into an HList of Expr nodes?
+sealed trait NonEmptyExprHList[F[_], V, T <: HList, P] {
```

Comment:
This is processed recursively using `EvalExprAsOutputAndParam` which calls back into `InterpretExprAsFunction` but strips off the `ExprResult` wrapper and returns the tuple of `(Output[R], P)`. This function is abstract enough that there is a `Functor` and `Semigroupal` definition, so the results can be mapped over to combine all the results back into an `HList`. This `HList` is then converted to a tuple is then re-wrapped using the `ExprResult.WrapOutput`.

Reply:
I removed this TODO. I don't think it is possible at this level.